Libs/Mesh/Mesh.cpp

```diff
@@ -1038,6 +1096,411 @@ std::ostream& operator<<(std::ostream &os, const Mesh& mesh)
+      vtkIdType ptid = locator->FindClosestPoint(ptdob);
+      mesh->GetPoint(ptid, ptdob);
+
+      // Add first point in boundary
+      if (i == 0) {
+        lastId = ptid;
+        Point3 pathpt;
+        pathpt = getPoint(ptid);
+        selectionPoints->InsertNextPoint(pathpt[0], pathpt[1], pathpt[2]);
+        boundaryVerts.push_back(ptid);
+      }
+      // If the current and last vetices are different, then add all vertices in the path to the boundaryVerts list
+      if (lastId != ptid) {
+        //std::cout << pt[0] << " " << pt[1] << " " << pt[2] << " -> " << ptdob[0] << " " << ptdob[1] << " " << ptdob[2] << std::endl;
+        // Add points in path
+        dijkstra->SetStartVertex(lastId);
+        dijkstra->SetEndVertex(ptid);
+        dijkstra->Update();
+        vtkSmartPointer<vtkIdList> idL = dijkstra->GetIdList();
+        for (size_t j = 1; j < idL->GetNumberOfIds(); j++) {
+          vtkIdType id = idL->GetId(j);
+          Point3 pathpt;
+          pathpt = getPoint(ptid);
+          selectionPoints->InsertNextPoint(pathpt[0], pathpt[1], pathpt[2]);
+          boundaryVerts.push_back(id);
+        }
+      }
+      lastId = ptid;
+    }
+
+    //std::cout << "Number of boundary vertices " << boundaryVerts.size() << std::endl;
+
+    vtkSmartPointer<vtkSelectPolyData> select = vtkSelectPolyData::New();
+    select->SetLoop(selectionPoints);
+    select->SetInputData(this->mesh);
+    select->GenerateSelectionScalarsOn();
+    select->SetSelectionModeToLargestRegion();
+
+    // Clipping mesh
+    vtkSmartPointer<vtkClipPolyData> selectclip = vtkClipPolyData::New();
+    selectclip->SetInputConnection(select->GetOutputPort());
+    selectclip->SetValue(0.0);
+
+    selectclip->Update();
+
+    MeshType halfmesh = selectclip->GetOutput();
+
+    vtkSmartPointer<vtkDoubleArray> inout = computeInOutForFFCs(query, halfmesh);
+
+    vtkSmartPointer<vtkDoubleArray> values = vtkSmartPointer<vtkDoubleArray>::New();
+    vtkSmartPointer<vtkDoubleArray> absvalues = setDistanceToBoundaryValueFieldForFFCs(values,
+                                                                                       points,
+                                                                                       boundaryVerts,
+                                                                                       inout, V, F,
+                                                                                       dom);
+
+    this->mesh->GetPointData()->SetActiveScalars("value");
+
+    std::vector<Eigen::Matrix3d> face_grad_ = this->setGradientFieldForFFCs(absvalues, V, F);
+
+  } // Per boundary for loop end
+
+  locator = vtkSmartPointer<vtkCellLocator>::New();
+  locator->SetDataSet(this->mesh);
+  locator->BuildLocator();
+
+  // Write mesh for debug purposes
+//    std::string fnin = "dev/mesh_" + std::to_string(dom) + "_" + std::to_string(num) + "_in.vtk";
+//    this->write(fnin);
+
+  return true;
+}
+
+// Copied directly from Meshwrapper
```

Comment:
Moving is difficult since both mesh and vtkmeshwrappers are wrappers for a vtkpolydata mesh. If I move it, I need to refactor out the whole of vtkmeshwrappers. This is why #1408 exists in the first place.